sqlglot/expressions.py

```diff
@@ -512,23 +514,39 @@ def from_arg_list(cls, args):
             args_dict[all_arg_keys[-1]] = args[arg_idx:]
         return cls(**args_dict)
 
+    @classmethod
+    def sql_names(cls):
+        if cls is Func:
+            raise NotImplementedError(
+                "SQL name is only supported by concrete function implementations"
+            )
+        if not hasattr(cls, "_sql_names"):
+            cls._sql_names = [camel_to_snake_case(cls.__name__)]
+        return cls._sql_names
+
     @classmethod
     def sql_name(cls):
-        if not hasattr(cls, "_sql_name"):
-            cls._sql_name = camel_to_snake_case(cls.__name__)
-        return cls._sql_name
+        return cls.sql_names()[0]
 
     @classmethod
     def default_parser_mappings(cls):
-        return {cls.sql_name: cls.from_arg_list}
+        return {name: cls.from_arg_list for name in cls.sql_names()}
+
+
+class AggFunc(Func):
```

Comment:
maybe just agg ?

Reply:
I don't mind but I'm a little concerned that in this case there is no way to tell that's this is actually a base class and not an actual function.

Reply:
should we use ABC?

Reply:
i'm fine to leave it aggfunc and no ABC, just a thought

Reply:
Got it, thanks. If your opinion on this is not super strong, I'd prefer to keep it `AggFunc` to make it stand out from concrete functions.